src/pybind/mgr/dashboard/module.py

```diff
@@ -68,6 +68,51 @@ def os_exit_noop(*args):
 
 logger = logging.getLogger(__name__)
 
+def cors_tool():
+    '''
+    Handle both simple and complex CORS requests
+
+    Add CORS headers to each response. If the request is a CORS preflight
+    request swap out the default handler with a simple, single-purpose handler
+    that verifies the request and provides a valid CORS response.
+    '''
+    req_head = cherrypy.request.headers
+    resp_head = cherrypy.response.headers
+
+    # Always set response headers necessary for 'simple' CORS.
+    cross_origin_url = mgr.get_module_option_ex('dashboard', 'cross_origin_url', None)
+    if cross_origin_url is not None:
+        resp_head['Access-Control-Allow-Origin'] = cross_origin_url
+    resp_head['Access-Control-Expose-Headers'] = 'GET, POST'
```

Comment:
Does it make sense to have a list of cross origin urls?

Reply:
Hmmm, it could be. Need to check if list i can pass list to Access-Control-Allow-Origin header. So far i only saw `*` and `origin`